
# HG changeset patch
# User Brendan Cully <brendan@kublai.com>
# Date 1175615951 25200
# Node ID e363d7a6904653f2b5acc17f6ea0da526bb63711
# Parent 943d0177c15449311ea2fa81efe8ecd700b06faa
Fix DB4 crash introduced in d5ab883ef90a

--- a/hcache.c	Tue Apr 03 08:44:24 2007 -0700
+++ b/hcache.c	Tue Apr 03 08:59:11 2007 -0700
@@ -992,6 +992,15 @@ mutt_hcache_open(const char *path, const
     FREE(&h);
     return NULL;
   }
+  ret = db_create (&h->db, h->env, 0);
+  if (ret)
+  {
+    h->env->close (h->env, 0);
+    mx_unlock_file (h->lockfile, h->fd, 0);
+    close (h->fd);
+    FREE (&h);
+    return NULL;
+  }
 
   if (stat(path, &sb) != 0 && errno == ENOENT)
   {

